vulnerabilities/importers/github.py

```diff
@@ -0,0 +1,342 @@
+
+        if pkg_name in self.cache:
+            return
+
+        artifact_comps = pkg_name.split(':')
+        endpoint = self.artifact_url(artifact_comps)
+        resp = requests.get(endpoint).content
+
+        try:
+
+            xml_resp = ET.ElementTree(ET.fromstring(resp.decode('utf-8')))
+            self.cache[pkg_name] = self.extract_versions(xml_resp)
+
+        except ET.ParseError:
+            self.cache[pkg_name] = set()
+
+    @staticmethod
+    def artifact_url(artifact_comps: List[str]) -> str:
+
+        base_url = 'https://repo.maven.apache.org/maven2/{}'
+        group_id, artifact_id = artifact_comps
+        group_url = group_id.replace('.', '/')
+        suffix = group_url + '/' + artifact_id + '/' + 'maven-metadata.xml'
+        endpoint = base_url.format(suffix)
+
+        return endpoint
+
+    @staticmethod
+    def extract_versions(xml_response: ET.ElementTree) -> Set[str]:
+
+        all_versions = set()
+        for child in xml_response.getroot().iter():
+            if child.tag == 'version':
+                all_versions.add(child.text)
+
+        return all_versions
+
+
+class NugetVersionAPI:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, pkg_name: str) -> Set[str]:
+        return self.cache.get(pkg_name.lower(), set())
+
+    def load_to_api(self, pkg_name: str) -> None:
+        if pkg_name in self.cache:
+            return
+        endpoint = self.nuget_url(pkg_name)
+        try:
+            resp = requests.get(endpoint).json()
+        # pkg_name=Microsoft.NETCore.UniversalWindowsPlatform triggers
+        # JSONDecodeError.
+        except json.decoder.JSONDecodeError:
+            self.cache[pkg_name.lower()] = set()
+            return
+
+        self.cache[pkg_name.lower()] = self.extract_versions(resp)
+
+    @staticmethod
+    def nuget_url(pkg_name: str) -> str:
+        base_url = 'https://api.nuget.org/v3/registration5-semver1/{}/index.json'
+        return base_url.format(pkg_name.lower())
+
+    @staticmethod
+    def extract_versions(json_resp: dict) -> Set[str]:
+        all_versions = set()
+        try:
+            for entry in json_resp['items'][0]['items']:
+                all_versions.add(entry['catalogEntry']['version'])
+        # json response for YamlDotNet.Signed triggers this exception
+        except KeyError:
+            return all_versions
+
+        return all_versions
+
+
+class ComposerVersionAPI:
+    def __init__(self):
+        self.cache = {}
+
+    def get(self, pkg_name: str) -> Set[str]:
+        return self.cache.get(pkg_name.lower(), set())
+
+    def load_to_api(self, pkg_name: str) -> None:
+        if pkg_name in self.cache:
+            return
+        endpoint = self.composer_url(pkg_name)
+        json_resp = requests.get(endpoint).json()
+        self.cache[pkg_name] = self.extract_versions(json_resp, pkg_name)
+
+    @staticmethod
+    def composer_url(pkg_name: str) -> str:
+        vendor, name = pkg_name.split('/')
```

Comment:
always prefer a more robust partition() or rpartition():
```
>>> vendor, name = pkg_name.split('/')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'pkg_name' is not defined
>>> vendor, _, name = 'foo'.partition('/')
>>> vendor, name
('foo', '')
>>> vendor, _, name = 'foo/bar'.partition('/')
>>> vendor, name
('foo', 'bar')
```

Reply:
>always prefer a more robust partition() or rpartition():

I didn't knew about this function. Thanks.
Robust in the sense it doesn't raise `ValueError: not enough values to unpack` .

 But in this case it makes more sense to me to use `split` because that would raise an exception and crash out VulnerableCode, instead of  ending up writing garbage in the DB.